receiver/stanzareceiver/factory.go

```diff
@@ -17,66 +17,59 @@ package stanzareceiver
 import (
 	"context"
 
-	stanza "github.com/open-telemetry/opentelemetry-log-collection/agent"
+	"github.com/open-telemetry/opentelemetry-log-collection/agent"
+	"github.com/open-telemetry/opentelemetry-log-collection/pipeline"
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/config/configmodels"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/receiver/receiverhelper"
 )
 
-const (
-	typeStr = "stanza"
-	verStr  = "0.12.0"
-)
-
-// NewFactory creates a factory for Stanza receiver
-func NewFactory() component.ReceiverFactory {
-	return receiverhelper.NewFactory(
-		typeStr,
-		createDefaultConfig,
-		receiverhelper.WithLogs(createLogsReceiver))
+// LogReceiverType is the interface used by stanza-based log receivers
+type LogReceiverType interface {
+	configmodels.Receiver
+	Version() string
+	CreateDefaultConfig() configmodels.Receiver
+	Decode(configmodels.Receiver) (pipeline.Config, error)
 }
 
-func createDefaultConfig() configmodels.Receiver {
-	return &Config{
-		ReceiverSettings: configmodels.ReceiverSettings{
-			TypeVal: configmodels.Type(typeStr),
-			NameVal: typeStr,
-		},
-	}
+// NewFactory creates a factory for a Stanza-based receiver
+func NewFactory(logReceiverType LogReceiverType) component.ReceiverFactory {
+	return receiverhelper.NewFactory(
+		logReceiverType.Type(),
+		logReceiverType.CreateDefaultConfig,
+		receiverhelper.WithLogs(createLogsReceiver(logReceiverType)),
+	)
 }
 
-// CreateLogsReceiver creates a logs receiver based on provided config
-func createLogsReceiver(
-	ctx context.Context,
-	params component.ReceiverCreateParams,
-	cfg configmodels.Receiver,
-	nextConsumer consumer.LogsConsumer,
-) (component.LogsReceiver, error) {
+func createLogsReceiver(logReceiverType LogReceiverType) receiverhelper.CreateLogsReceiver {
+	return func(
+		ctx context.Context,
+		params component.ReceiverCreateParams,
+		cfg configmodels.Receiver,
+		nextConsumer consumer.LogsConsumer,
+	) (component.LogsReceiver, error) {
+		pipeline, err := logReceiverType.Decode(cfg)
+		if err != nil {
+			return nil, err
+		}
 
-	obsConfig := cfg.(*Config)
+		emitter := NewLogEmitter(params.Logger.Sugar())
+		converter := NewConverter(string(logReceiverType.Type()), logReceiverType.Version())
```

Comment:
Similarly, there is no expectation that file log receiver will populate the instrumentation library name. I do not think it is right to set the field to the receiver type.

Reply:
Ok, I'll remove this. I was thinking of this with a "something is better than nothing" mindset, but will actually be much cleaner since the conversion code can be stateless.